### PR TITLE
Cut release-0.7

### PR DESCRIPTION
**Pull request by dgrisonnet**
Cut release 0.7 of kube-prometheus.

Once merged, @s-urbaniak will finalize the release.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -94,19 +94,20 @@ $ minikube addons disable metrics-server
...
-| `release-0.3`         | ✔               | ✔               | ✔               | ✔               | ✗               | ✗               |
-| `release-0.4`         | ✗               | ✗               | ✔ (v1.16.5+)    | ✔               | ✗               | ✗               |
-| `release-0.5`         | ✗               | ✗               | ✗               | ✗               | ✔               | ✗               |
-| `release-0.6`         | ✗               | ✗               | ✗               | ✗               | ✔               | ✔               |
-| `HEAD`                | ✗               | ✗               | ✗               | ✗               | x               | ✔               |
+| kube-prometheus stack | Kubernetes 1.14 | Kubernetes 1.15 | Kubernetes 1.16 | Kubernetes 1.17 | Kubernetes 1.18 | Kubernetes 1.19 | Kubernetes 1.20 |
+|-----------------------|-----------------|-----------------|-----------------|-----------------|-----------------|-----------------|-----------------|
+| `release-0.3`         | ✔               | ✔               | ✔               | ✔               | ✗               | ✗               | ✗               |
```
Comment (dgrisonnet) [inline README.md:99]: Should we keep supporting `release-0.3`?

**Comment by s-urbaniak** [comment]
cc @prometheus-operator/kube-prometheus-reviewers

File: README.md (hunk 1 above)
Comment (lilic) [reply README.md:99]: Lets remove the mention of release-0.3 yes, we will still support it. This is compatibility rather than support scheme.

I would stick to last 5 releases (including master) and last 5 k8s versions.

File: jsonnet/kube-prometheus/jsonnetfile.json [hunk 2]
```diff
@@ -44,7 +44,7 @@
           "subdir": ""
         }
       },
-      "version": "master"
+      "version": "release-0.6"
```
Comment (lilic) [inline jsonnet/kube-prometheus/jsonnetfile.json:47]: These we can just do as a PR to the release branch after its cut, otherwise here in master you need to undo the changes, as we always want master to point to master.

**Review by lilic — COMMENTED** [review]
Thanks!

I would also like us to bump k8s version to test against 1.20 as well, should be out now. WDYT?

**Comment by dgrisonnet** [comment]
@lilic I already updated the CI to test against both 1.19 and 1.20, did you had anything else in mind?

File: jsonnet/kube-prometheus/jsonnetfile.json (hunk 2 above)
Comment (dgrisonnet) [reply jsonnet/kube-prometheus/jsonnetfile.json:47]: @lilic I guess I should also do the same for kube-state-metrics?

**Comment by lilic** [comment]
No, just did not show up for me before, all good, thanks! 👍 🎉

File: jsonnet/kube-prometheus/jsonnetfile.json (hunk 2 above)
Comment (lilic) [reply jsonnet/kube-prometheus/jsonnetfile.json:47]: kube-state-metrics alerts mixin is okay to be master or pinned, either work as we did not change anything there, but we can keep it pinned to 1.9 as well. Important is to not change ksm own mixin to master, as we have breaking changes in master.

File: README.md (hunk 1 above)
Comment (dgrisonnet) [reply README.md:99]: Ok, I'll remove the 1.14 and 1.15 rows. However, since we don't have any version fully compatible with 1.16, is it meaningful to keep release-0.3 for now?

Comment (lilic) [reply README.md]: > Lets remove the mention of release-0.3 yes, we will still support it. 
> I would stick to last 5 releases (including master) and last 5 k8s versions.

Yes as mentioned, I believe 5 versions master being one of them, removes the release-0.3 no?

File: jsonnet/kube-prometheus/jsonnetfile.json [hunk 3]
```diff
@@ -35,7 +35,7 @@
           "subdir": "jsonnet/mixin"
         }
       },
-      "version": "master"
+      "version": "release-0.44"
```
Comment (lilic) [inline jsonnet/kube-prometheus/jsonnetfile.json:38]: Sorry for not pointing out, this can also stay at master and can just be done as a PR towards the release branch, otherwise we need to change it twice as in master it will always stay master. If that makes sense?

Comment (dgrisonnet) [reply jsonnet/kube-prometheus/jsonnetfile.json]: Makes sense, sorry I misunderstood your first comment about this. It should be good now.

**Comment by dgrisonnet** [comment]
The E2E tests seems to be failing because prometheus-operator v0.44.1 has not been fully released yet.

**Review by lilic — APPROVED** [review]
lgtm on green, thanks!
